File: LeetCode/10/lc1033.py
# lc1033.py

# LeetCode 1033 Moving Stones Until Consecutive `E`
# A~0c08
# 1sk | 90%


class Solution:
    def numMovesStones(self, a: int, b: int, c: int) -> List[int]:
        if a < 1 or a > 100 or b < 1 or b > 100 or c < 1 or c > 100:
            return [-1, -1]

        a, b, c = sorted([a, b, c])

        minMoves = -1
        if a + 1 == b and b + 1 == c:
            minMoves = 0
        elif a + 1 == b or b + 1 == c or a + 2 == b or b + 2 == c:
            minMoves = 1
        else:
            minMoves = 2

        return [minMoves, c - a - 2]

File: LeetCode/10/lc1033.c
/* lc1033.c */

/* LeetCode 1033 Moving Stones Until Consecutive `E` */
/* A~0c08 */
/* 1sk | 100% | 24' */

int *numMovesStones(int a, int b, int c, int *returnSize)
{
    int *ret = (int *)calloc(sizeof(int), 2);
    ret[0] = ret[1] = -1;
    *returnSize = 2;

    if (a < 1 || a > 100 || b < 1 || b > 100 || c < 1 || c > 100)
        return ret;

    if (a > b)
    {
        int tmp = a;
        a = b;
        b = tmp;
    }
    if (a > c)
    {
        int tmp = a;
        a = c;
        c = tmp;
    }
    if (b > c)
    {
        int tmp = b;
        b = c;
        c = tmp;
    }

    if (a + 1 == b && b + 1 == c)
    {
        ret[0] = ret[1] = 0;
        return ret;
    }
    else if (a + 1 == b || b + 1 == c || a + 2 == b || b + 2 == c)
        ret[0] = 1;
    else
        ret[0] = 2;

    ret[1] = c - a - 2; /* = (b - a - 1) + (c - b - 1) */

    return ret;
}
